Makefile

```diff
@@ -159,6 +159,15 @@ else
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test $(GO_FLAGS) ./apis/... ./pkg/... ./test/yq/... ./controllers/... -covermode=atomic -coverprofile coverage.out
 endif
 
+.PHONY: vector-test
+vector-test: ## Run vector tests
+	@echo Generating test files for Vector tests
+	$(eval TMP := $(shell mktemp -d))
+	VECTOR_TEST_FILES=true OUTPUT_PATH=$(TMP) go test -v ./pkg/telemetry -run=TestGenerateTomlTestFiles
+	@echo Running vector test files
+	OUTPUT_PATH=$(TMP) VECTOR=$(VECTOR) scripts/run-vector-tests.sh
```

Comment:
[thought] Couldn't we just run the `vector` executable (and assert its return value) from within the Go test?
That way it would be a regular test, we wouldn't need a dedicated make target / script file / CI step to run it.

Reply:
We could do that also, but I don't think we've done that too often in our tests. It's an external program, so not really running go test code. 

But would we really want Vector installation to be a dependency on running Go tests? ``make test`` would require then vector to be always installed.